Book/php7/internal_types/functions/callables.rst

```diff
@@ -0,0 +1,140 @@
+PHP Callables
+===================
+
+Dealing with PHP functions in C requires the knowledge of the following two structures
+``zend_fcall_info``/``zend_fcall_info_cache``. The first one necessarily contains the information for calling
+the function, such as arguments and the return value, but may also include the actual callable.
+The latter *only* contains the callable. We will use the commonly used abbreviation of FCI and FCC when talking about
+``zend_fcall_info`` and ``zend_fcall_info_cache`` respectively.
+You will most likely encounter those when using the ZPP ``f`` argument flag, or when you need to call a PHP function
+or method from within an extension.
+
+Structure of ``zend_fcall_info``
+--------------------------------
+
+.. warning:: The implementation of ``zend_fcall_info`` is widely different prior to PHP 7.1.0.
+
+As of PHP 8.0.0, ``zend_fcall_info`` has the following structure:
+
+::
+
+    struct _zend_fcall_info {
+        size_t size;
+        zval function_name;
+        zval *retval;
+        zval *params;
+        zend_object *object;
+        uint32_t param_count;
+        /* This hashtable can also contain positional arguments (with integer keys),
+         * which will be appended to the normal params[]. This makes it easier to
+         * integrate APIs like call_user_func_array(). The usual restriction that
+         * there may not be position arguments after named arguments applies. */
+        HashTable *named_params;
+    } zend_fcall_info;
+
+
+Let detail the various FCI fields:
+
+``size``:
+  Mandatory field, which is the size of an FCI structure, thus always: ``sizeof(zend_fcall_info)``
+``function_name``:
+  Mandatory field, the actual callable, do not be fooled by the name of this field as this is a leftover when
+  PHP didn't have objects and class methods. It must be a string zval or an array following the same rules as
+  callables in PHP, namely the first index is a class or instance object, and the second one is the method name.
+  It can also be undefined if, and only if, a non empty FCC is provided.
+``retval``:
+  Mandatory field, which will contain the result of the PHP function
+``param_count``:
+  Mandatory field, the number of arguments that will be provided to this call to the function
+``params``:
+  contains positional arguments that will be provided to this call to the function.
+  If ``param_count = 0``, it can be ``NULL``.
+``object``:
+  The object on which to call the method name stored in ``function_name``, or ``NULL`` if no objects are involved.
+``named_params``:
+  A HashTable containing named (or positional) arguments.
+
+.. warning:: Prior to PHP 8.0.0, the ``named_params`` field did not exist. However, a ``zend_bool no_separation;``
+   field existed which specified if array arguments should be separated or not.
```

Comment:
Hopefully I am correct in my assumption in what this field did.

Reply:
I think that was only relevant for by-ref parameters, but even if so, we can fix that later. However, I'd use `.. note::` instead of `.. warning::` here.